ui/src/app/pages/workspace/workspace-card.tsx

```diff
@@ -171,32 +175,13 @@ export const WorkspaceCard = fp.flow(withNavigation)(
         : triggerEvent(EVENT_CATEGORY, 'navigate', 'Click on workspace name');
     }
 
-    onClick() {
-      const {
-        workspace: { id, namespace },
-      } = this.props;
+    onClick(e) {
       if (this.requiresReviewPrompt()) {
         this.setState({ showResearchPurposeReviewModal: true });
-      } else {
-        this.trackWorkspaceNavigation();
-        this.props.navigate(['workspaces', namespace, id, 'data']);
+        e.preventDefault();
       }
     }
 
-    getRightClickTarget() {
```

Comment:
This might have just been an evolution of an existing method name, but as a rule of thumb, if you start talking about explicitly handling ctrl-click, right-click, or meta-clicks in the code, you're probably headed down the wrong path (unless you're doing something very unusual). There will generally be a more fundamental browser mechanism for getting the behavior you are looking for. In this case, using `e.preventDefault()` is a reasonable way to interrupt the normal behavior of the anchor element.

Reply:
I understand that, however the reason i had this method (rt-click) is to restrict the options on workspace care name to show open in new tab etc,  if the workspace requires review prompt. If i do not do that, choosing option 'Open in  A new tab' say would just open the workspace in a new tab without any prompt.

Reply:
OK, I see the intent now. The research review feature is supposed to have a guard, which was removed per RW-6975. In this scenario it would redirect to the research purpose review in a new tab - I think that's sufficient to handle an edge case like this.